src/main/java/com/github/ygimenez/method/Pages.java

```diff
@@ -130,52 +130,44 @@ public static MessageHandler getHandler() {
 	 */
 	public static void paginate(@Nonnull Message msg, @Nonnull List<Page> pages) throws ErrorResponseException, InsufficientPermissionException {
 		if (!isActivated()) throw new InvalidStateException();
-		List<Page> pgs = Collections.unmodifiableList(pages);
+		List<Page> pageList = Collections.unmodifiableList(pages);
+		List<Emote> reactions = List.of(PREVIOUS, CANCEL, NEXT);
 		clearReactions(msg);
-
-		msg.addReaction(paginator.getEmotes().get(PREVIOUS)).submit();
-		msg.addReaction(paginator.getEmotes().get(CANCEL)).submit();
-		msg.addReaction(paginator.getEmotes().get(NEXT)).submit();
+		addReactions(msg, reactions);
 
 		handler.addEvent(msg, new ThrowingBiConsumer<>() {
-			private final int maxP = pgs.size() - 1;
-			private int p = 0;
-			private final Consumer<Void> success = s -> {
+			private final int lastPageIndex = pageList.size() - 1;
+			private int pageIndex = 0;
+			private final Consumer<Void> successConsumer = s -> {
 				handler.removeEvent(msg);
 				if (paginator.isDeleteOnCancel()) msg.delete().submit();
 			};
 
 			@Override
-			public void acceptThrows(@Nonnull User u, @Nonnull GenericMessageReactionEvent event) {
-				Message m = null;
+			public void acceptThrows(@Nonnull User user, @Nonnull GenericMessageReactionEvent event) {
+				Message message;
 				try {
-					m = event.retrieveMessage().submit().get();
+					message = event.retrieveMessage().submit().get();
 				} catch (InterruptedException | ExecutionException ignore) {
-				}
-				if (u.isBot() || m == null || !event.getMessageId().equals(msg.getId()))
 					return;
+				}
 
-				MessageReaction.ReactionEmote reaction = event.getReactionEmote();
-				if (checkEmote(reaction, PREVIOUS)) {
-					if (p > 0) {
-						p--;
-						Page pg = pgs.get(p);
-
-						updatePage(m, pg);
-					}
-				} else if (checkEmote(reaction, NEXT)) {
-					if (p < maxP) {
-						p++;
-						Page pg = pgs.get(p);
+				if (user.isBot() || !event.getMessageId().equals(msg.getId()))
+					return;
 
-						updatePage(m, pg);
+				MessageReaction.ReactionEmote userReaction = event.getReactionEmote();
+				reactions.forEach(reaction -> {
```

Comment:
forEach is really, really bad to use and should be avoided whenever possible (always prefer normal for-each or Stream API)

Reply:
Crap yeah good shout, I thought that used stream internally. Will change